### PR TITLE
Prow: generic comment handling

### PR DESCRIPTION
**Pull request by cjwagner**
This PR switches plugins that handle comment text to use the generic comment event type instead of registering for all event types that could contain text.
Previously if a plugin wanted to recognize a command `/do-it`, it would need to register for `IssueComment`, `Issue`, `PullRequest`, `Review`, and `ReviewComment` events in order to detect the command anywhere on github. This resulted in many plugins implementing a generic `event` struct and coercing each event type into an `event` so that processing could be implemented generically on `event`s. Now plugins can register for `GenericCommentEvent`s and the various event types associated with comments will be coerced to the generic type before being passed to the plugin.

/cc @Q-Lee @rmmh

### CODE REVIEW

File: prow/plugins/sigmention/sigmention.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "sigmention"
+
+var (
+	sigMatcher = regexp.MustCompile(`(?m)@kubernetes/sig-([\w-]*)-(misc|test-failures|bugs|feature-requests|proposals|pr-reviews|api-reviews)`)
```
Comment (kargakis) [inline prow/plugins/sigmention/sigmention.go]: Can the org flow from config?

**Comment by spxtr** [comment]
Could you split the label plugin stuff into a separate PR? +847 −825 is hard to reckon with.

**Comment by cjwagner** [comment]
PTAL

**Pull request merged**